src/sql/src/plan/error.rs

```diff
@@ -231,6 +232,23 @@ pub enum PlanError {
     // TODO(benesch): eventually all errors should be structured.
     Unstructured(String),
 }
+#[derive(Clone, Debug)]
+pub enum ConnectionParsingError {
```

Comment:
Why is this a separate type? We don't have any other example of splitting out `PlanError` within this crate.